### PR TITLE
Improved rate limiting tests

### PR DESCRIPTION
**Pull request by MadL1me**
# Improved rate limiting middleware tests

Added rate limiting middleware tests improvements, based on #43298 issue

## Description

- Unit tests now using `RateLimiterEndpointConventionBuilderExtensions ` where applicable
- Added `Assert-Act-Arrange` comments for more clarification
- Extracted method for rate limit middleware initialization
- Extracted build classes in another file

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/aspnetcore?pullRequest=45541) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/aspnetcore?pullRequest=45541) before we can accept your contribution.<br/><br/>:x: MadL1me [sign now](https://cla.dotnetfoundation.org/dotnet/aspnetcore?pullRequest=45541)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/aspnetcore?pullRequest=45541) it.</sub>

File: src/Middleware/Middleware.slnf [hunk 1]
```diff
@@ -125,6 +125,8 @@
       "src\\Servers\\Kestrel\\Kestrel\\src\\Microsoft.AspNetCore.Server.Kestrel.csproj",
       "src\\Servers\\Kestrel\\Transport.Quic\\src\\Microsoft.AspNetCore.Server.Kestrel.Transport.Quic.csproj",
       "src\\Servers\\Kestrel\\Transport.Sockets\\src\\Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets.csproj",
+      "src\\Testing\\src\\Microsoft.AspNetCore.Testing.csproj",
+      "src\\Testing\\test\\Microsoft.AspNetCore.Testing.Tests.csproj",
```
Comment (MadL1me) [inline src/Middleware/Middleware.slnf:129]: When I was settuping my dev environment with Visual Studio, I had a lot of issues with unresolved dependencies. After adding this two projects to filter, the issue disappeared. I don't know it that right thing to commit thought

Comment (Tratcher) [reply src/Middleware/Middleware.slnf:129]: Adding Testing is fine. Testing.Tests shouldn't be needed, nothing should depend on it.

VS has a feature where you can right-click on a project and tell it to load all transitive dependencies.

File: src/Middleware/RateLimiting/test/EndpointBuilder.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.RateLimiting;
+
+internal class TestEndpointBuilder : EndpointBuilder
```
Comment (wtgodbe) [inline src/Middleware/RateLimiting/test/EndpointBuilder.cs:14]: Please rename this file to `TestEndpointBuilder.cs`

File: src/Middleware/Middleware.slnf (hunk 1 above)
Comment (wtgodbe) [reply src/Middleware/Middleware.slnf:129]: Agreed, please remove Microsoft.AspNetCore.Testing.Tests.csproj

**Review by wtgodbe — CHANGES_REQUESTED** [review]
Other than the 2 small comments, looks great! Thanks for the change. Will approve & merge once those 2 changes have been made.

**Review by wtgodbe — APPROVED** [review]

**Comment by wtgodbe** [comment]
Looks great! Will merge once CI passes.

**Pull request merged**